pandas/core/groupby/ops.py

```diff
@@ -583,7 +583,11 @@ def _transform(
         return result
 
     def agg_series(self, obj: Series, func):
-        if is_extension_array_dtype(obj.dtype) and obj.dtype.kind != "M":
+        if len(obj) == 0:
+            # SeriesGrouper would raise if we were to call _aggregate_series_fast
```

Comment:
Hmm should we not just fix `_aggregate_series_fast` to not raise in that case then? Hoping to avoid special casing like this in any groupby functions

Reply:
> Hoping to avoid special casing like this in any groupby functions

Agreed on the goal.  ATM this is the best way to make progress towards that goal.  In particular, making this explicit here is much clearer than catching the ValueError with the particular message on L600.

> should we not just fix _aggregate_series_fast to not raise in that case then? 

#29499 does something along those lines.